enterprise/cmd/frontend/internal/authz/webhooks/github.go

```diff
@@ -0,0 +1,159 @@
+package webhooks
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	gh "github.com/google/go-github/v43/github"
+
+	"github.com/sourcegraph/log"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/webhooks"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/repoupdater"
+	"github.com/sourcegraph/sourcegraph/internal/repoupdater/protocol"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+var githubEvents = []string{
+	"repository",
+	"member",
+	"organization",
+	"membership",
+	"team",
+	"team_add",
+}
+
+type GitHubWebhook struct {
+	logger log.Logger
+}
+
+func NewGitHubWebhook(logger log.Logger) *GitHubWebhook {
+	return &GitHubWebhook{logger}
```

Comment:
Hey, it's me, Nit van Pick

```suggestion
	return &GitHubWebhook{logger: logger}
```